### PR TITLE
(chore) O3-3551: Use latest Actions in GH Actions

### PR DESCRIPTION
**Pull request by Bcoderx6**
https://openmrs.atlassian.net/browse/O3-3551

This pull request updates the GitHub Actions workflow to utilize the latest versions of several key actions, improving performance, security, and feature set. The specific updates include:

1.actions/checkout: Updated from `v2` to `v4`.
2. actions/cache: Updated from `v2` to `v3`.
3.actions/setup-java: Updated from `v1` to `v3`.

Benefits:
- Enhanced Performance: The latest versions come with performance improvements, which can speed up the workflow execution.
- Improved Security: Newer versions include the latest security patches, reducing the risk of vulnerabilities.
- Advanced Features: The updated actions provide additional features and improvements, such as better caching mechanisms and more robust setup options for Java.

Detailed Changes:
1. actions/checkout@v4
   - Improved performance and reliability in fetching the repository.
   - Enhanced support for submodules and large repositories.

2. actions/cache@v3
   - Optimized caching strategies to reduce build times.
   - Better handling of cache keys and restore-keys, ensuring more efficient cache usage.

3. actions/setup-java@v3
   - Better support for multiple Java versions and configurations.
   - Improved caching of Java dependencies, reducing setup times.

### CODE REVIEW

**Comment by Bcoderx6** [comment]
@jayasanka-sack! , Kindly review the updates made to the GitHub Actions workflow.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -21,13 +21,13 @@ jobs:
       JAVA_VERSION: ${{ matrix.java-version }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
```
Comment (jayasanka-sack) [inline .github/workflows/maven.yml:26]:
```suggestion
        uses: actions/setup-java@v4
```

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -21,13 +21,13 @@ jobs:
...
         with:
           java-version: ${{ matrix.java-version }}
       - name: Cache local Maven repository
-        uses: actions/cache@v2
+        uses: actions/cache@v3
```
Comment (jayasanka-sack) [inline .github/workflows/maven.yml:30]:
```suggestion
        uses: actions/cache@v4
```

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Thanks for the PR, @Bcoderx6 ! Could you please address the requested changes?

**Comment by Bcoderx6** [comment]
@jayasanka-sack ! Fine, I've done those changes.

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -21,13 +21,13 @@ jobs:
       JAVA_VERSION: ${{ matrix.java-version }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v4
```
Comment (jayasanka-sack) [inline .github/workflows/maven.yml:26]:
```suggestion
        uses: actions/setup-java@v4
        distribution: 'zulu'
```

File: .github/workflows/maven.yml [hunk 4]
```diff
@@ -21,13 +21,14 @@ jobs:
...
+        distribution: 'zulu'
         with:
           java-version: ${{ matrix.java-version }}
```
Comment (jayasanka-sack) [inline .github/workflows/maven.yml:29]:
```suggestion
        with:
          java-version: ${{ matrix.java-version }}
          distribution: 'zulu'
```

File: .github/workflows/maven.yml [hunk 5]
```diff
@@ -21,13 +21,14 @@ jobs:
...
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v4
         with:
           java-version: ${{ matrix.java-version }}
+          distribution: 'zulu'
```
Comment (jayasanka-sack) [inline .github/workflows/maven.yml:29]:
```suggestion
          distribution: 'temurin'
```

**Pull request merged**